common/scala/src/main/scala/whisk/core/entity/ArtifactActivationStore.scala

```diff
@@ -0,0 +1,128 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package whisk.core.entity
+
+import java.time.Instant
+
+import akka.actor.ActorSystem
+import akka.stream.ActorMaterializer
+
+import spray.json.JsObject
+
+import whisk.common.{Logging, TransactionId}
+import whisk.core.database.{ArtifactStore, CacheChangeNotification, StaleParameter}
+
+import scala.concurrent.Future
+import scala.util.{Failure, Success}
+
+class ArtifactActivationStore(actorSystem: ActorSystem, actorMaterializer: ActorMaterializer, logging: Logging)
+    extends ActivationStore {
+
+  implicit val executionContext = actorSystem.dispatcher
+
+  private val artifactStore: ArtifactStore[WhiskActivation] =
+    WhiskActivationStore.datastore()(actorSystem, logging, actorMaterializer)
+
+  def store(activation: WhiskActivation)(implicit transid: TransactionId,
+                                         notifier: Option[CacheChangeNotification]): Future[DocInfo] = {
+
+    logging.debug(this, s"recording activation '${activation.activationId}'")
+
+    val res = WhiskActivation.put(artifactStore, activation)
+
+    res onComplete {
+      case Success(id) => logging.debug(this, s"recorded activation")
+      case Failure(t) =>
+        logging.error(
+          this,
+          s"failed to record activation ${activation.activationId} with error ${t.getLocalizedMessage}")
+    }
+
+    res
+  }
+
+  def get(activationId: ActivationId)(implicit transid: TransactionId): Future[WhiskActivation] = {
+    WhiskActivation.get(artifactStore, DocId(activationId.asString))
+  }
+
+  def delete(activationId: ActivationId)(implicit transid: TransactionId,
```

Comment:
Might want to add a comment saying that this is not used in production/performance relevant code. (since the get/delete cycle is kinda overheady)